week-9/Homework/mandatory/2-exercises/4-music-player.js

```diff
@@ -18,36 +18,59 @@ This means the order the songs are played in will be random, but each song will
 
  */
 
-
 class MusicPlayer {
-    // Add your code here
-
+  constructor() {
+    this.i = 0;
+    this.playList = [];
+  }
+  add(music, singer) {
+    this.playList.push(`${music} by ${singer}`);
+  }
+  play() {
+    if (this.i > this.playList.length || this.i < 0) {
+      console.log(`there is no song to play`);
+    } else {
+      console.log(`Currently Playing: ${this.playList[this.i]}`);
+    }
+  }
+  skip() {
+    if (this.i < this.playList.length - 1) {
+      this.i++;
+      this.play();
+    } else {
+      console.log(`There isn't any song to play`);
+    }
+  }
+
+  previous() {
+    this.i--;
+    if (this.i < 0) {
+      this.i = this.playList.length - 1;
+    }
+    this.play();
+  }
+
+  shuffle() {
+    let random = Math.floor(Math.random() * this.playList.length);
```

Comment:
How would you make sure that the song won't be repeated when `shuffle` is called multiple times in a row?